_posts/2025-08-26-strimzi-metrics-reporter.md

```diff
@@ -0,0 +1,85 @@
+---
+layout: post
+title: "Using Strimzi Metrics Reporter to Expose Kafka Metrics in Prometheus Format"
+date: 2025-10-01
+author: Owen Corrigan
+---
+When deploying new Kafka cluster it may be one of your intentions to monitor the cluster using metrics.
+`Apache Kafka®` generates and exposes metrics that reflect how a system is operating. These metrics are 
+useful for monitoring, troubleshooting, tuning, and capacity planning when it comes to running your Kafka cluster.
+In essence, monitoring is crucial to ensure the health and performance of your Kafka clusters.
+`Prometheus` as a metrics monitoring solution has become increasingly common due to its popularity in the cloud-native ecosystem and it is this format that we use to expose metrics.
+
+The [Strimzi Metrics Reporter](https://github.com/strimzi/metrics-reporter) project implements the Java `MetricsReporter` interface to expose metrics via `Prometheus` and it is available from Strimzi v0.48.0.
+
+Up to this point, we have recommended using the `JMX Exporter` as a way to expose Kafka metrics, and we have provided some example yaml files in our [Strimzi Examples folder](https://github.com/strimzi/strimzi-kafka-operator/tree/main/packaging/examples/metrics).
+However, using the JMX Exporter adds unnecessary complexity. Metrics reporters first expose metrics through JMX, and then a Java agent re-exposes them over HTTP.
+The `Strimzi Metrics Reporter` exposes metrics in the Prometheus format directly over HTTP, simplifying the setup and reducing resource consumption.
```

Comment:
Should we also mention that this is a `MetricsReporter` plugin that can be used with any Kafka deployment?